common/djangoapps/student/views.py

```diff
@@ -2329,6 +2329,9 @@ def password_reset_confirm_wrapper(
                 entry = PasswordHistory()
                 entry.create(updated_user)
 
+            updated_user.backend = 'django.contrib.auth.backends.ModelBackend'
```

Comment:
honestly, not sure. i took it from the superuser_login_as method. it doesn't work otherwise.